### DISCUSSION TITLE
Recommended approach for enhancing generated OpenAPI documentation

### DISCUSSION THREAD

**Discussion by kjcsb1** [PostgREST/postgrest#3815] [Q&A] [answered]
PostgREST 12.2

As [recommended](https://docs.postgrest.org/en/v12/explanations/schema_isolation.html), we expose views (not tables) through PostgREST and those are included in the automatically-generated OpenAPI documentation.

I would like to:
1. include view-level and column-level comments. Comments added to tables (as recommended in the [documentation](https://docs.postgrest.org/en/v12/references/api/openapi.html) ) don't appear in the generated OpenAPI file
2. order and group the generated paths. The generated OpenAPI file doesn't appear to be in any particular or admin-configurable order

What are the recommendations for generating documentation? One option would be to maintain a list of views in a table containing metadata. We could then write a script to retrieve the metadata and enhance the generated file.

Any suggestions appreciated.

**Comment by wolfgangwalther** [comment, marked as answer]
> include view-level and column-level comments.

I just tested and comments on views and columns of views appear in the OpenAPI output:

```sql
COMMENT ON VIEW <schema>.<view> IS 'hello';
COMMENT ON COLUMN <schem>.<view>.<column> IS 'world';
```

> order and group the generated paths.

That's currently not possible.

**Reply by kjcsb1** [reply]
Well that's embarrassing! [Updated the docs](https://github.com/PostgREST/postgrest/pull/3816) to make that clear.

**Reply by steve-chavez** [reply]
> order and group the generated paths.
> That's currently not possible.

@laurenceisla Is that possible to do in a standard way? (Maybe for https://github.com/PostgREST/postgrest-openapi)
